src/custom-element.js

```diff
@@ -767,6 +767,8 @@ function createBaseAmpElementProto(win) {
     if (this.isInTemplate_) {
       return;
     }
+    this.implementation_.viewportCallback(false);
+    this.implementation_.unlayoutCallback();
```

Comment:
I don't like `#unlayoutCallback` being called by default. I think it's too heavy-handed, let the component decide how to release it's resources with `#detachedCallback`.